windows-cxx17.cmake

```diff
@@ -0,0 +1,25 @@
+# Copyright (c) 2016-2018, Ruslan Baratov
+# Copyright (c) 2017, David Hirvonen
+# All rights reserved.
+
+if(DEFINED POLLY_WINDOWS_CXX17_CMAKE_)
+  return()
+else()
+  set(POLLY_WINDOWS_CXX17_CMAKE_ 1)
+endif()
+
+include("${CMAKE_CURRENT_LIST_DIR}/utilities/polly_init.cmake")
+
+polly_init(
+    "clang / c++17 support"
+    "MinGW Makefiles"
+)
+
+add_compile_definitions(
+    WIN32_LEAN_AND_MEAN
+    _WIN32_WINNT=0x0A00 
+)
+set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}  -pthread -femulated-tls " CACHE STRING "")

```

Comment:
Instead of forcing -pthread we could fix the FindThreads.cmake module to support our case as nxxm does enable threads in the cmakelists.txt.

Reply:
Is it possible to switch to true Thread Local Storage ? Do you have links to the fixes pending in clang.

Reply:
pthread is not necesseray. we need it when we use clang parameter in boost when we use gcc that's good 

and for -femulated-tls : link : 
https://stackoverflow.com/questions/31617034/stdfuture-and-clang-with-stdlib-libstdc
msys2/MINGW-packages#5786